File: apps/user-app-web/src/lib/api/locationApi.ts
import { apiClient } from './client';
import { extractSingle, extractList, type BffResponse } from './bffParser';

export interface RegionInfo {
  regionType: string;
  addressName: string;
  region1: string;
  region2: string;
  region3: string;
  region4: string;
  code: string;
}

export interface NearbyClub {
  id: number;
  name: string;
  location: string;
  address: string;
  phone: string;
  latitude: number | null;
  longitude: number | null;
  totalHoles: number;
  totalCourses: number;
  status: string;
  clubType: string;
  facilities: string[];
  distance: number; // km
}

export const locationApi = {
  reverseGeo: async (lat: number, lon: number): Promise<RegionInfo | null> => {
    const response = await apiClient.get<BffResponse<RegionInfo>>(
      '/api/user/location/reverse-geo',
      { lat, lon },
    );
    return extractSingle<RegionInfo>(response.data);
  },

  nearbyClubs: async (
    lat: number,
    lon: number,
    radius?: number,
    limit?: number,
  ): Promise<NearbyClub[]> => {
    const params: Record<string, number | undefined> = { lat, lon };
    if (radius) params.radius = radius;
    if (limit) params.limit = limit;

    const response = await apiClient.get<BffResponse<NearbyClub[]>>(
      '/api/user/location/nearby-clubs',
      params,
    );
    return extractList<NearbyClub>(response.data);
  },
};

File: marketplace/apps/user-app-android/app/src/main/java/com/parkgolf/app/data/remote/api/LocationApi.kt
package com.parkgolf.app.data.remote.api

import com.parkgolf.app.data.remote.dto.common.ApiResponse
import com.parkgolf.app.data.remote.dto.location.NearbyClubDto
import com.parkgolf.app.data.remote.dto.location.RegionInfoDto
import retrofit2.http.GET
import retrofit2.http.Query

interface LocationApi {

    @GET("api/user/location/reverse-geo")
    suspend fun reverseGeo(
        @Query("lat") lat: Double,
        @Query("lon") lon: Double
    ): ApiResponse<RegionInfoDto>

    @GET("api/user/location/nearby-clubs")
    suspend fun nearbyClubs(
        @Query("lat") lat: Double,
        @Query("lon") lon: Double,
        @Query("radius") radius: Double = 30.0,
        @Query("limit") limit: Int = 10
    ): ApiResponse<List<NearbyClubDto>>
}
